File: C11/백준/Gold/1806. 부분합/부분합.c
#include <stdio.h>
#define INF 1000001

int main() {
    int n, s;
    scanf("%d %d", &n, &s);

    int arr[n];
    for (int i = 0; i < n; i++) {
        scanf("%d", &arr[i]);
    }

    int sum = 0, start = 0, end = 0, minLength = INF;

    while (1) {
        if (sum >= s) {
            if (end - start < minLength)
                minLength = end - start;
            sum -= arr[start++];
        } else if (end == n) {
            break;
        } else {
            sum += arr[end++];
        }
    }

    if (minLength == INF)
        printf("0\n");
    else
        printf("%d\n", minLength);

    return 0;
}

File: Java/백준/Gold/1806. 부분합/부분합.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.io.IOException;
import java.util.StringTokenizer;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        int n = Integer.parseInt(st.nextToken());
        int s = Integer.parseInt(st.nextToken());

        int[] arr = new int[n];
        st = new StringTokenizer(br.readLine());
        for (int i = 0; i < n; i++) {
            arr[i] = Integer.parseInt(st.nextToken());
        }

        int minLength = Integer.MAX_VALUE;
        int sum = 0;
        int start = 0;

        for (int end = 0; end < n; end++) {
            sum += arr[end];

            while (sum >= s) {
                minLength = Math.min(minLength, end - start + 1);
                sum -= arr[start];
                start++;
            }
        }

        System.out.println(minLength == Integer.MAX_VALUE ? 0 : minLength);
    }
}